### PR TITLE
Support to f4 address and general improvements for filecoin integration

### PR DESCRIPTION
**Pull request by emmanuelm41**
### 📝 Description

We have removed addresses related logic from Ledger Live, and replaced it by the [Izari-Filecoin](https://github.com/Zondax/izari-filecoin) package. It makes it easy for developers to incorporate Filecoin the project. By doing so, we have added support to new address type 4, which has been released on mainnet. 

Besides, we apply some general improvements in order to avoid possible bug in certain scenarios.

Now, address validation and serialization is done by Izari-Filecoin. The package has support to f4/t4 addresses. 

**Notes**
- PR reopened. Prev PR: https://github.com/LedgerHQ/ledger-live/pull/2372

### ❓ Context

- **Impacted projects**: `` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/families/filecoin/bridge/account.ts [hunk 1]
```diff
@@ -142,8 +142,8 @@ const estimateMaxSpendable = async ({
 
   if (balance.lte(estimatedFees)) return new BigNumber(0);
 
-  balance.minus(estimatedFees);
-  if (amount) balance.minus(amount);
+  balance = balance.minus(estimatedFees);
+  if (amount) balance = balance.minus(amount);
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/filecoin/bridge/account.ts:146]: Good catch, could you consider this as an opportunity to unit test this function?
We use jest, you'll find plenty of examples in the codebase, including ways to mock functions and backend calls.

File: libs/ledger-live-common/src/families/filecoin/bridge/utils/addresses.ts [hunk 2]
```diff
@@ -1,150 +1,126 @@
...
-type PrefixType = keyof PrefixObjType;
+// @ts-ignore
+import BN from "bn.js";
 
-type ProtocolsObjType = Readonly<
-  { "0": "ID" } & { "1": "SECP256K1" } & { "2": "Actor" } & {
-    "3": "BLS";
-  }
->;
+// @ts-ignore
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/filecoin/bridge/utils/addresses.ts:7]: When using a `@ts-ignore`, please add the code of the specific error that's ignored (https://github.com/Microsoft/TypeScript/blob/main/src/compiler/diagnosticMessages.json)
A comment to justify is welcome too.

**Review by haammar-ledger — COMMENTED** [review]
Hi @emmanuelm41 thanks for these improvements.

Could you please update the description of this PR with a brief test plan covering your changes?
For our QA to know what should be checked.

**Comment by emmanuelm41** [comment]
@haammar-ledger could you please give it a look again? I just made a "big" change by using a new package we created, extracting much of the logic we copied here.

File: libs/ledger-live-common/src/families/filecoin/bridge/account.ts (hunk 1 above)
Comment (emmanuelm41) [reply libs/ledger-live-common/src/families/filecoin/bridge/account.ts:146]: I will try to add a new test case for this hammar

**Comment by haammar-ledger** [comment]
Hi @emmanuelm41, as discussed, I advise you use `proto` to solve the issues that are currently present in the CI.

**Comment by emmanuelm41** [comment]
@haammar-ledger it is fixed now!

**Review by haammar-ledger — COMMENTED** [review]
@emmanuelm41 I don't quite get it but there are still errors related to `node` version in the CI, e.g:
- https://github.com/LedgerHQ/ledger-live/actions/runs/4631707390/jobs/8194905822#step:8:3389
May be on our side, I'll check and let you know.

Also these errors which I don't quite get either, but both seem related to the new `@zondax/izari-filecoin` lib you're using:
- https://github.com/LedgerHQ/ledger-live/actions/runs/4631706815/jobs/8194902651#step:4:31
- https://github.com/LedgerHQ/ledger-live/actions/runs/4631707185/jobs/8194903524#step:14:8738

File: libs/ledger-live-common/dependencies.md
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply libs/ledger-live-common/dependencies.md:1]: Done!

**Comment by emmanuelm41** [comment]
@haammar-ledger please, you can check again. Unfortunately, Jest is required to enable an experimental feature in other to accept dynamic imports [🔗 ](https://github.com/Zondax/izari-filecoin/blob/98d9a3e5d1e692bc52d60e67624298e6aa271b06/package.json#L71-L84). That is something we have internally on our package in order to keep ESM and CJS compatibility (some of our dependencies are ESM only). Now it is fixed. Please, check it again!

**Comment by emmanuelm41** [comment]
I have another question. BigInt are not yet supported on ReactNative, are it? I am not sure if it has been supported or not yet.

File: libs/ledger-live-common/dependencies.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-This document allows to track, explain and maintain the dependencies we have defined in package.json.
+This document allows to track, ex.json.plain and maintain the dependencies we have defined in package
```
Comment (haammar-ledger) [inline libs/ledger-live-common/dependencies.md:1]: There seems to be a stray `.json.` here?

**Review by haammar-ledger — CHANGES_REQUESTED** [review]
- A couple lint errors, see here: https://github.com/LedgerHQ/ledger-live/actions/runs/4645540430/jobs/8221433635#step:8:715
- Some issue was fixed this morning on our CI regarding the build for mobile, you can rebase on develop and try again.

File: libs/ledger-live-common/dependencies.md (hunk 3 above)
Comment (emmanuelm41) [reply libs/ledger-live-common/dependencies.md]: Done!

**Comment by emmanuelm41** [comment]
@haammar-ledger done! I have applied requested changes.

**Comment by haammar-ledger** [comment]
Thanks!

Sorry I missed your question:
> I have another question. BigInt are not yet supported on ReactNative, are it? I am not sure if it has been supported or not yet.

ReactNative actually supports BigInt since a few versions, but until now BigInt wasn't supported on the Live codebase because we needed to update our RN version. However this was just merged this very morning :slightly_smiling_face: https://github.com/LedgerHQ/ledger-live/pull/2888 it's a coincidence but we should now support BigInt.
Please note that the codebase wasn't migrated to use it yet, so the standard way around this remains to use the _bignumber.js_ library.

**Comment by emmanuelm41** [comment]
@haammar-ledger I don't want to do anything on Ledger Live with BigInt. But we had some dependencies that uses it internally. We needed to make fork and remove BigInt references from it in order to be able to integrate Filecoin. If BigInt is supported now, we can go back to the original package. That is a great news!!! 

Let me understand this. This means we could use that package back again, doesn't it?

**Review by haammar-ledger — APPROVED** [review]
@emmanuelm41 New conflicts to solve :pray: otherwise LGTM!

**Comment by emmanuelm41** [comment]
> @emmanuelm41 New conflicts to solve 🙏 otherwise LGTM!

Let me rebase it @haammar-ledger

**Comment by gre** [comment]
👋  ESLint config has been unified on develop after we merged https://github.com/LedgerHQ/ledger-live/pull/3622 – this will improve the project consistency, remove eslint dependencies duplicates, and overall simplify the maintenance. This comes with a prettier library upgrade and unification of the prettierrc too, so you likely have conflicts with develop.


Hopefully, here is the procedure to easily fix this. **please proceed as follow**:

**Step 1: sync your branch with develop up to the config changes:**

2 cases

- if your branch were continuously rebased with develop and don't have any merge commit, you can do 
```
git rebase aa1e0ebc0a0498b4350754899b6a368fbd42e811
```
- otherwise prefer 
```
git merge aa1e0ebc0a0498b4350754899b6a368fbd42e811
```

**then, apply the config**

```
pnpm i
pnpm lint:fix --force
```

if there are remaining errors that lint couldn't fix, please fix them and `pnpm lint` again.

then commit in your branch.

```
git commit -a -m "chore: lint fixes"
```

**Step 2: finalize the rebase/merge with develop**

2 cases

- if your branch were continuously rebased with develop and don't have any merge commit, you can do
```
git pull --rebase origin develop
```
- otherwise prefer 
```
git merge origin/develop
```

you should now be up to date. 👍

> *Thanks and sorry for the inconvenience,*
> *gre*

**Pull request closed without merging**